app/assets/javascripts/components/admin_notes/notes_modal_trigger.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Cookies } from 'react-cookie-consent';
+
+const NotesModalTrigger = ({ setIsModalOpen, notesList, setNoteFetchTimestamp }) => {
+  const [notificationCount, setNotificationCount] = useState(0);
+
+  const onClickAdminButton = () => {
+    setIsModalOpen('adminNotePanel');
+    setNotificationCount(0);
+    setNoteFetchTimestamp();
+  };
+
+  useEffect(() => {
+    // Retrieve the last fetch timestamp from the cookie
+    const userLastFetchTimestamp = Cookies.get('lastFetchAdminNoteTimestamp');
+
+    // Check if the cookie value is a valid number
+    const parsedTimestamp = isFinite(userLastFetchTimestamp) ? parseInt(userLastFetchTimestamp) : 0;
+
+    // Calculate the count of new notes
+    const newAdminNoteCount = notesList.filter(note => new Date(note.updated_at) > new Date(parsedTimestamp)).length;
+
+    // Update the notification count state
+    setNotificationCount(newAdminNoteCount);
+  }, [notesList]);
+
+  // Accessible message for screen readers, indicating whether there are new admin notes or none available
+  const notesAriaLabel = !notesList.length ? I18n.t('notes.admin.aria_label.no_notes_available')
+  : I18n.t('notes.admin.aria_label.new_notes_message', { count: notificationCount, plural: notificationCount > 1 ? 's' : '' });
```

Comment:
This is not a good way to handle plurals, as it will just add 's' no matter the language.

I suggest rewriting it to just take the count parameter, such as by using "note(s)" in the message.

Reply:
Okay